### ISSUE TITLE
Question: incompatibility with `.phpstorm.meta.php`?

### ISSUE THREAD

**Issue by boesing** [Lctrs/psalm-psr-container-plugin#352]
#### Steps required to reproduce the problem

1. Create a `.phpstorm.meta.php` directory
2. Create a `test.meta.php` and add an override for `ContainerInterface::get(0)` like below
3. Execute psalm with psr-container-plugin enabled

**.phpstorm.meta.php/test.meta.php**
```php
<?php
/**
 * @see https://confluence.jetbrains.com/display/PhpStorm/PhpStorm+Advanced+Metadata
 */

namespace PHPSTORM_META {
    use Psr\Container\ContainerInterface;
    use Qoo;

    override(ContainerInterface::get(0), map([
        'qoo' => Qoo::class,
    ]));
}
```

**Example file**
```php
<?php

class Qoo
{
}

class Baz
{
    public const QOO = 'qoo';
}

/** @var \Psr\Container\ContainerInterface */
$container = null;

$foo = $container->get(Baz::qoo);
/** @psalm-trace $foo */
```

#### Expected Result

* `$foo` is detected as instance of `Qoo`

#### Actual Result

* `$foo` is detected as instance of `Baz` (`::class` constant was not used, maybe another bug here?)

----

Psalm has `.phpstorm.meta.php` compatibility: https://github.com/vimeo/psalm/blob/466137a4c42b55b24a9ca588bf395f9f29d4b004/src/Psalm/Config.php#L1969

**Comment by Lctrs** [comment]
Hi !

It's not an incompatibility with phpstorm meta per se. You just unveil a bug in the plugin, our argument resolving logic seems to be totally wrong. I'm currently rewriting the logic and will provide a fix in the next days.

**Comment by boesing** [comment]
Oh wow, did not expect anything to change soon. Please take your time - no need to focus on this imho. If I can be of any help, let me know. 😊

**Comment by Lctrs** [comment]
May you test #355 please ?

**Comment by boesing** [comment]
Sorry for not responding. I am currently quite busy and thus had no time. I will check this this afternoon.
Anyways: thanks for working on this!
